src/lib/mongoBackend/MongoCommonUpdate.cpp

```diff
@@ -3150,6 +3173,8 @@ void processContextElement
     return; // Error already in responseP
   } 
 
+  cacheSemTake(__FUNCTION__, "Processing Context Element");
```

Comment:
Yeah, I know that the 'sem-lock-span' has grown a lot.
To be protected, it actually has to grow.
However, we might be able to narrow it down a little.
Let's talk ...